### PR TITLE
feat(header): add Header component

### PR DESCRIPTION
**Pull request by nathanjamal**
### Description

* feat(header): add Header component

created a Header component. Incomplete at the moment and lacking proper documentation

HEYUI-283

* feat(header): control lang select menu with keyboard

allow language select menu to be opened and closed with keyboard + added some aria data

HEYUI-283

* feat(header): added keyboard support

Added keyboard support, Aria labels and some tests for the hooks and utils

HEYUI-283

* feat(header): added navigation component

added navigation component that builds a nav and subnav based on an object

HEYUI-283

* feat(header): burger menu functionality

added burger menu functionality and moved some functions out of view files

HEYUI-283

* feat(header): mobile search open and close functionality

added toggle on and off for mobile search

HEYUI-283

* docs(header): move header into organisms folder

Merge main and put header under organisms in the storybook

HEYUI-283

* feat(header): move header items that are not visible at mobile into the burger menu

language select and account are not visible in the header in mobile breakpoints. so we move them
into the nav menu

HEYUI-283

* change to single semi-colon syntax

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change

### How do I test this

- Add steps to test
- in bullet point format
- preferably you can add link to the storybook build in the PR

### Did you remember to take care of the following?

- [x] `npm i` – for new NPM dependencies.
- [x] `npm run lint` - to check for linting issues
- [x] `npm run test` - to run unit tests
- [x] `npm run test:sh:docker` - to run screenshot tests, [detail instruction](https://hey-car.github.io/heycar-uikit/main/?path=/docs/guidelines-screenshot-testing--page)

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing.
- [x] Add new passing unit tests to cover the code introduced by your pr.

Thanks for contributing!

### CODE REVIEW

**Comment by nathanjamal** [comment]
NOTE: The jest coverage report is complaining about a few things in the header component. The things that it has brought up are covered, but in the screenshot tests. The Nature of this component means screenshot tests are a better way to test a lot of the interactivity

File: packages/header/src/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      ? getCurrentLang(langItemConfig!.currentLang, langItemConfig!.options)
+      : undefined;
+    const LangIco = currentLang?.icon;
+    const { isLangListOpen, keyboardOpen, setIsFocused, setIsHovering } =
+      useLangList();
+
+    // Event handlers
+    const handleSearchToggle = (newState: boolean) => {
+      setIsSearchOpen(newState);
+      if (searchItemConfig?.onClick) searchItemConfig.onClick(newState);
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:83]: Toggling the search is covered in the screenshot tests

File: packages/header/src/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                    aria-label={`${locale.langListHeading} - ${locale.spaceBarNotification}`}
+                    className={`${styles.horizontalNavOnly} ${styles.item} ${
+                      isLangListOpen ? styles.focused : ''
+                    }`}
+                    onBlur={() => setIsFocused(false)}
+                    onFocus={() => setIsFocused(true)}
+                    onKeyDown={e => keyboardOpen(e)}
+                    onMouseOut={() => setIsHovering(false)}
+                    onMouseOver={() => setIsHovering(true)}
+                  >
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:177]: Most of these are covered in the screenshot tests

File: packages/header/src/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                  </button>
+                  {isLangListOpen && (
+                    <LanguageList
+                      dataTestId="header-language-list"
+                      heading={locale.langListHeading}
+                      onFocusEvents={setIsFocused}
+                      onHoverEvents={setIsHovering}
+                      options={langItemConfig!.options}
+                      trackingFn={trackingFn}
+                    />
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:191]: Language select dropdown covered in the screenshot tests

File: packages/header/src/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+              {/*
+               *** Call Button
+               */}
+              {hasCall && (
+                <Link
+                  aria-label={callItemConfig!.label}
+                  className={`${styles.desktopOnly} ${styles.item} ${styles.asideItem}`}
+                  href={callItemConfig!.href}
+                  onClick={() =>
+                    itemOnClick(
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:225]: covered in headerItemHelper.test.tsx

File: packages/header/src/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+                </Link>
+              )}
+              {/*
+               *** Mobile menu Toggle
+               */}
+              <button
+                aria-label={DEFAULT_LOCALE.burgerMenuButtonLabel}
+                className={`${styles.notHorizontalNav} ${styles.item}`}
+                onClick={() =>
+                  itemOnClick('Toggle Menu', trackingFn, () =>
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:245]: covered in headerItemHelper.test.tsx

File: packages/header/src/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+                    setIsNavTrayOpen(!isNavTrayOpen),
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:246]: Burger menu toggle covered in screenshot tests

File: packages/header/src/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+                  )
+                }
+              >
+                {isNavTrayOpen ? <Close /> : <Hamburger />}
+              </button>
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:251]: Burger menu toggle covered in screenshot tests

File: packages/header/src/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+            </Grid.Col>
+          </Grid.Row>
+        </Grid.Container>
+        {/*
+         *** Navigation
+         */}
+        <div
+          className={`${styles.navWrapper} ${isNavTrayOpen ? 'navOpen' : ''}`}
+        >
```
Comment (nathanjamal) [inline packages/header/src/Header.tsx:260]: burger menu toggle covered in screenshot tests

**Review by esraltintas — APPROVED** [review]

**Pull request merged**